demo

```diff
@@ -1,4 +1,5 @@
 #!/bin/bash
+set -e
```

Comment:
Unsure about this one. It now exits when one of the curl calls fails, and then I guess the background docker-compose process will be killed also.

Reply:
I think it won't actually. I think maybe we should resort to manual error checking for this file. The artifact already being uploaded is a very real possibility, if the environment is restarted, and this should be handled gracefully, with a message like "already uploaded" or something.

Reply:
Yes, I agree. However, this brings us to the log-frequency of the containers. Most likely this will never be seen by an end-user, as the containers have already spewed out several hundred lines of logs. Any good ideas on this one? Could docker compose write to a file? and then after the initial setup (curls and wgets ) is done, then we can `tail -f  ` the file or something?

Reply:
Hmm, good question. My initial idea for this implementation was to do this uploading in a container of its own, which would enable you to view its logs separately by using `docker logs`. But that is a more complicated solution and I kind of like the simplicity of your approach as well. It is possible to call `docker-compose up -d` which defers it to the background, including the logs. Later you can "resume" viewing the logs with `docker-compose logs -f`. But the great disadvantage with this method is that Ctrl-C stops working, and you can no longer stop the demo environment by pressing that combo. All you will do then is interrupt the log viewing.

A couple of suggestions, none of which are particularly good:

* Use the `docker-compose up -d` trick, but implement a `trap SIGINT` in the shell code to catch Ctrl-C and call `docker-compose down` manually.
* Simply accept that the log will scroll away together with the rest
* Move the code to an execute-and-exit container